CorsixTH/Lua/audio.lua

```diff
@@ -206,6 +204,9 @@ function Audio:initSpeech(speech_file)
     archive_data = load_sound_file(speech_file)
   end
 
+  self.sound_archive = nil
+  self.sound_fx = nil
+
```

Comment:
I think we should be setting speech_file_name to nil here too? Otherwise it could error out below then not load again above.

Reply:
The not_loaded variable is fishy - why aren't we setting it on all the not_loaded conditions but only there?

Reply:
Added speech_file_name, couldn't hurt. Not sure I follow the not_loaded comment, unless you mean it just looks odd, but I assume as its driven from that global audio, and it only sets it to true, if there is no music either loaded, which is done in Audio:init. No music, no sounds, disable it.